### PR TITLE
[ANALYZER-3220] - Impossible to add into dashboard analyzer report wi…

### PR DESCRIPTION
**Pull request by apratkin**
…th prompt that contains special chars.

### CODE REVIEW

**Comment by apratkin** [comment]
@pentaho-nbaker could you please review my commit?

**Comment by mchen-len-son** [comment]
@rmansoor @rfellows please take a look.

**Comment by mchen-len-son** [comment]
@rmansoor @rfellows @bennychow could you please take a look?
Thanks.

**Comment by bennychow** [comment]
With this change, at least you can open the Analyzer report in the dashboard... however, does the prompt with special characters in its name actually work?  I.e. can you create a dashboard prompt and hook that up to the Analyzer filter prompt?

File: core/src/org/pentaho/platform/util/beans/BeanUtil.java [hunk 1]
```diff
@@ -103,8 +103,12 @@ public Object getValue( String propertyName ) throws IllegalAccessException, Inv
    * @return <code>true</code> if the bean property can be written to
    */
   public boolean isWriteable( String propertyName ) {
-    return propUtil.isWriteable( bean, propertyName )
-        || ( propUtil.getResolver().isIndexed( propertyName ) && propUtil.isReadable( bean, propertyName ) );
+    try {
+      return propUtil.isWriteable( bean, propertyName )
+          || ( propUtil.getResolver().isIndexed( propertyName ) && propUtil.isReadable( bean, propertyName ) );
+    } catch ( IllegalArgumentException exception ) {
+      return false;
```
Comment (mbatchelor) [inline core/src/org/pentaho/platform/util/beans/BeanUtil.java]: I think there should be some logging here - at least debug-level logging. This is swallowing the exception completely.

**Comment by apratkin** [comment]
@bennychow 
I tested with following steps:
1.	Rename some measure to ‘
2.	Add this measure
3.	Add some columns
4.	Filter by measure ‘
5.	Rename some column to ‘’ and add parametrized filter with that column(I did not found if we could create parametrized filter for measure)
6.	Added create report to new dashboard
7.	Change value of filter parameter

**Comment by apratkin** [comment]
@bennychow 
I tested with following steps:
1.	Create new analysis report
2.	Create filter ‘contains’ with named parameter ‘[~!@#$%^&*(){}|.,]-=_+|;'"?<>~`:’ and the same value.
3.	Create new dashboard and add report. Verify no data displayed because of filter (it is expected)
4.	Change filter value to some normal value and verify it shows some result.

**Comment by apratkin** [comment]
@mbatchelor 
About logging. I suppose it make sense add debug level logging. 
Inside PropertyUtilsBean. isWriteable method we have following code:
            try {
                nestedBean = getProperty(bean, next);
            } catch (IllegalAccessException e) {
                return false;
            } catch (InvocationTargetException e) {
                return false;
            } catch (NoSuchMethodException e) {
                return false;
            }

Based on this I decided do not log this exception.

**Comment by apratkin** [comment]
Updated commit with debug logging

**Comment by apratkin** [comment]
@pedrofvteixeira could you please take a look?

**Comment by apratkin** [comment]
@pamval could you please take a look?

**Comment by bennychow** [comment]
+1 Looks good to me now

**Comment by mchen-len-son** [comment]
Checked locally.
Merging, since no objections were raised after the PR was updated.

**Pull request merged**